extendr-api/src/conversions/try_into_int.rs

```diff
@@ -27,25 +28,27 @@ macro_rules! impl_into_integerish {
     ($float_type:ty, $int_type:ty) => {
         impl FloatToInt<$int_type> for $float_type {
             fn try_into_int(&self) -> Result<$int_type, ConversionError> {
-                if *self == <$float_type>::default() {
-                    return Ok(<$int_type>::default());
+                match self.classify() {
+                    FpCategory::Nan | FpCategory::Subnormal => Err(ConversionError::NotIntegerish),
```

Comment:
I think this is correct. Can you explain it to me though?

Reply:
Neither `NaN` nor a subnormal float are representable by integers (unless you want to default to `0`). It is quote challenging to get a subnormal float manually, so if we get one, I prefer to fail. 
Don't forget, these conversions are designed to handle things like indices, where people call it as `my_func(x, 4)`. And `4` is actual `4.0f64` and we want to cast it to `4i32` or `4usize`. Every other case should be a hard error

Reply:
Alright. Good enough for me, and for now